### DISCUSSION TITLE
Ignore Own Sub-Modules In Java Renovate

### DISCUSSION THREAD

**Discussion by JoaoCosme** [renovatebot/renovate#32781] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Hello!

We have a Java project with several sub-modules within it, each having their own pom.xml files and it's own versioning. As part of our pipeline, after building this project it's released with a timestamp of the time of deploy, so, for example, an artifact would be 1.0.0-20241128-<timestamp-of-deploy> instead of only 1.0.0. This is causing renovate to consider the sub-modules as dependencies to be updated. 

Is it possible to ignore all of these sub-modules dependencies? Currently we are using the ignoreDeps option, but this did not allow the usage of regex to add all sub-modules at once, so we would have extra effort every time we add a new sub-module.

Thanks for your help!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Yes, use a package rule matching against that project or project patterns, and use `allowedVersions` to control which pattern of versions are allowed (using regex)
